### DISCUSSION TITLE
Any other way to get RouteId from inside a middleware?

### DISCUSSION THREAD

**Discussion by goenning** [dotnet/yarp#576] [Q&A] [answered]
I found that the RouteId is assigned to **context.GetEndpoint.DisplayName** and it works fine for my use case, which is to perform some transformation and validation based on that, but it does look brittle at the moment as it could change in upcoming versions. Is there any other way to get access to the RouteId (or even RouteConfig) using the HttpContext? Thanks.

**Comment by Tratcher** [comment, marked as answer]
What kind of transformations did you have in mind? We've got a whole infrastructure for that.
https://microsoft.github.io/reverse-proxy/articles/transforms.html

Edit: And no, that's the only way to get the route id right now.

**Reply by goenning** [reply]
we're looking to do request/response body transformation which **will** vary per route, hence why the custom middleware and why it's important for us to know which route is being proxied for a particular request

**Reply by Tratcher** [reply]
Makes sense. Do you actually inspect the route id, or do you use that as a key to look up the metadata you need to make this decision? I ask because there's a few different ways that custom information could be appended to a route and exposed to middleware.

E.g. ProxyRoute.Metadata
https://github.com/microsoft/reverse-proxy/blob/c81cf0bc0a65dd5c79f4b7015f7fb8a679d9adb4/src/ReverseProxy/Abstractions/RouteDiscovery/Contract/ProxyRoute.cs#L53,

Or Endpoint Metadata: https://github.com/microsoft/reverse-proxy/pull/328

**Reply by goenning** [reply]
Yes, that was my initial plan. I was going to use the RouteId to look up the Metadata from the list of proxy routes, which is what triggered this thread looking for better ways of retrieving the RouteId. Is there any other way to have access to the ProxyRoute.Metadata from a middleware? I saw that RouteConfig is stored in the Endpoint, but unfortunately it's unaccessible. Thanks

**Reply by Tratcher** [reply]
ProxyRoute.Metadata isn't exposed at the moment, but that does sound like what you'd need here. Can you open something at https://github.com/microsoft/reverse-proxy/issues to track that?

**Reply by goenning** [reply]
Thanks, just opened this one :+1: https://github.com/microsoft/reverse-proxy/issues/587
